### PR TITLE
Fixed  `Realm not defined` error under Jest

### PR DESCRIPTION
**Pull request by fronck**
## What, How & Why?

Fixed error that would cause `Realm.Set`'s iterators to throw a `Realm not defined` error.
This closes RJS-1196.

## ☑️ ToDos

* [x] 📝 Changelog entry
~* [ ] 📝 `Compatibility` label is updated or copied from previous entry~
~* [ ] 🚦 Tests~
~* [ ] 📝 Public documentation PR created or is not necessary~
~* [ ] 💥 `Breaking` label has been applied or is not necessary~

*If this PR adds or changes public API's:*
~* [ ] typescript definitions file is updated~
~* [ ] jsdoc files updated~
~* [ ] Chrome debug API is updated if API is available on React Native~

### CODE REVIEW

File: lib/collection-methods.js [hunk 1]
```diff
@@ -55,7 +55,7 @@ Object.defineProperty(iteratorPrototype, Symbol.iterator, {
 
 ['entries', 'keys', 'values'].forEach(function(methodName) {
     var method = function() {
-        const isSet = this instanceof Realm.Set;
+        const isSet = (this.constructor && this.constructor.name && this.constructor.name === 'Set');
```
Comment (kneth) [inline lib/collection-methods.js:58]: I would suggest that we add a comment why `instanceof Realm.Set` doesn't work.

**Review by kneth — APPROVED** [review]

File: lib/collection-methods.js (hunk 1 above)
Comment (kraenhansen) [reply lib/collection-methods.js:58]: Wouldn't it be better to repeat the pattern you introduced for `set-methods.js`? https://github.com/realm/realm-js/blob/ca7051f8dc566101a8a89444575d16d5167ed83f/lib/set-methods.js#L21-L30

Basically refactoring `collections-methods.js` to export a function that takes the `Realm` constructor as an argument.

**Comment by elliottkember** [comment]
I also had this error with v10.5 while running React Native under a debugger. I was able to fix it by adding `global.Realm = Realm` to a root file, but this would solve it in a nicer way.

**Review by takameyer — APPROVED** [review]

**Comment by liamjones** [comment]
Was this fix abandoned/unneeded? I see the branch was deleted without merging.

**Comment by takameyer** [comment]
@liamjones This branch was removed do to some merge conflicts, but the problem should have been fixed.  Can you make a new issue with some steps to reproduce `Realm not defined`?  We would appreciate it 😃

**Comment by liamjones** [comment]
Sure, it's happening in one of our app monorepo projects so I'll see if I can reduce it down to a minimal repro.

**Comment by Acetyld** [comment]
Can this be fixed asap?

**Pull request closed without merging**